boilerplate/openshift/golang-osd-operator/csv-generate/common-generate-operator-bundle.py

```diff
@@ -69,43 +203,81 @@
             "version": crd["spec"]["version"]
         }
     )
+# These will be written to the bundle at the end along with generic resources
 
-csv['spec']['install']['spec']['clusterPermissions'] = []
+## Process [Cluster]Role[Binding]s (TODO: Match up ServiceAccounts)
+# Role and ClusterRole are processed similarly
+rolekind_csvkey_bindingmap = (
+    ('ClusterRole', 'clusterPermissions', crb_by_cr),
+    # NOTE: The schema supports `permissions` for Roles, but OLM may not. For
+    # backward compatibility, treat Roles and RoleBindings as generic bundle
+    # resources.
+    # ('Role', 'permissions', rb_by_role),
+)
+for kind, csvkey, binding_map in rolekind_csvkey_bindingmap:
+    if kind in by_kind:
+        csv['spec']['install']['spec'][csvkey] = []
+    for role in by_kind.get(kind, []):
+        log_resource(role)
+        # Figure out the ServiceAccount.
+        binding = binding_map.get(role['metadata']['name'], None)
+        sa = discover_service_account(binding)
+        # Get rid of the binding, if found, so it doesn't end up in generic
+        # bundle resources.
+        trim_index(by_kind, f"{kind}Binding", binding)
+        # Inject the rules and ServiceAccount
+        csv['spec']['install']['spec'][csvkey].append(
+            {
+                'rules': role['rules'],
+                'serviceAccountName': sa,
+            })
+    # Get rid of these so we can iterate over what's left at the end
+    trim_index(by_kind, kind, 'ALL')
```

Comment:
I'm not a 100% sure but OLM may support provisioning `ClusterRoles` for service accounts in other namespaces not associated with the operator. I'm certain it does so for `Roles`. So I would recommend only trimming CRs and CRBs which are associated with the operator service account.

Reply:
We're already integrating all the CRs and CRBs by folding them into the CSV. The schema does not have a field for ServiceAccount namespace, so I don't see how that could be supported. But you raise a good point about the Roles and RoleBindings: At the moment we're just putting those in the bundle but ignoring their ServiceAccounts. I imagine we need to put those in the bundle as well.

I wonder if redundant ServiceAccounts (i.e. listed in a `clusterPermissions` entry and also in a bundle file) hurt anything. Anyway, I can go ahead with the work to pare down all SAs that we've already "used" and just bundle the rest.

Reply:
(Later) On further thought & discussion, I don't believe we're ever creating ServiceAccounts in other namespaces -- so let's leave this alone unless you find a counterexample.

Reply:
From what I can see on cluster and other operators OLM doesn't create ServiceAccounts in namespaces other than the operators install namespace.